Sc 30756 Create Superhub

Anakhom opened the pull request:
[sc-30756-add-learning-superhub-page-to-qml](https://app.shortcut.com/xanaduai/story/30756/add-learning-superhub-page-to-qml)
[figma](https://www.figma.com/file/DDc3TX8eUZ4ccXzj7mqE4q/PennyLane-Refresh?node-id=10%3A162&t=47hSkAuxkn3GqCVK-0)

- Moved Videos and Demos to new section
- Replaces current content with placeholder lorem ipsum
- Refactored HTML for better semantics 
- Added content
- All subhubs and new pages are merged in 

Final demo with content and new graphics:


https://user-images.githubusercontent.com/58717051/215874108-0f824443-d96e-4276-80a0-b38daea1b35f.mov

AndrewGardhouse commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
-            This, in turn, is driving the development of quantum machine learning and variational quantum circuits.</p>
+    <div>
+        <h1 class="mt-6">Welcome to the Quantum learning </br> superhub</h1>
+        <p class="lead grey-text text-center mx-auto my-5">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Curabitur tempus urna at turpis condimentum lobortis.</p>
```
@josh146 @BM7878 Do we have copy for this?

BM7878 replied:
I'm not sure where the best place to put this is, so I'll just put it here for now:

`
<h1>Learn how to program for quantum computers using PennyLane</h1>

<p>We have entered a new era. Quantum devices are becoming more and more advanced. Their usage is becoming a more and more active field of research. </p>

<p>Get started in programming for quantum computers by choosing your area of interest below.</p>
`

josh146 replied:
```suggestion
        <p class="lead grey-text text-center mx-auto my-5">We have entered a new era. Quantum devices are becoming more and more advanced. Their usage is becoming a more and more active field of research.<br><br>
        Get started in programming for quantum computers by choosing your area of interest below, or check out our collection of demos and tutorials.</p>
```

AndrewGardhouse commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
+                                Quantum </br> Computing
+                            </h3>
+                            <p class="card-text">
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the. 
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
```
@josh146 @BM7878 Do we have copy for this?

BM7878 replied:
As per above:

`
<h3>Quantum Computing</h3>

<p>A field of research that explores how universal computation can be achieved through quantum-mechanical systems.</p>

`

josh146 replied:
```suggestion
                            A field of research that explores how universal computation can be achieved through quantum-mechanical systems.
```


@BM7878 alternatively we could go with 

> Explore general quantum computing concepts and algorithms, from quantum volume to boson sampling.

AndrewGardhouse commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
+                                Quantum machine learning
+                            </h3>
+                            <p class="card-text">
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the. 
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
```
@josh146 @BM7878 Do we have copy for this?

BM7878 replied:
As per above:

```html

<h3>Quantum Machine Learning</h3>

<p>A field of research looking at the interplay of ideas from quantum computing and machine learning.</p>
```

josh146 replied:
```suggestion
                            Find out how the principles of quantum computing and machine learning can be united to create something new.
```

@BM7878 modified it slightly here

AndrewGardhouse commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
+                                Quantum Chemistry
+                            </h3>
+                            <p class="card-text">
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the. 
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
```
@josh146 @BM7878 Do we have copy for this?

BM7878 replied:
As per above:

```html
<h3>Quantum Chemistry</h3>

<p>A field of research focused on calculating the properties of molecules, and the materials made from them, using quantum mechanics.</p>
```

josh146 replied:
@BM7878 another small modification to focus on QC:

```suggestion
                            Study the properties of molecules and materials using quantum computing.
```

AndrewGardhouse commented on index.rst:
```diff
@@ -102,61 +109,108 @@ Quantum machine learning
+                                Demos
+                            </h3>
+                            <p class="card-text">
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the. 
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
```
@josh146 @BM7878 Do we have copy for this?

BM7878 replied:
For the first part:

```html
<h2>Browse our full set of demonstrations and tutorials</h2>
```

BM7878 replied:
For the second part:

```html
<h3>Demos</h3>

<p>Dive into the code and see how to apply cutting-edge algorithms and techniques with PennyLane.</p>
```

josh146 replied:
```suggestion
                            Dive into the code and see how to apply cutting-edge algorithms and techniques with PennyLane.
```

AndrewGardhouse commented:
Converting this to a draft PR since we are missing new copy and we don't want to accidentally merge this in an incomplete state. CC @BM7878 @josh146

BM7878 commented on index.rst:
```diff
@@ -102,61 +109,108 @@ Quantum machine learning
+                                Videos
+                            </h3>
+                            <p class="card-text">
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the. 
+                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
```
As per above:

```html
<h3>Videos</h3>

<p>Follow along with our experts and explore ideas in quantum programming, quantum machine learning, and quantum chemistry.</p>
```

josh146 replied:
```suggestion
                            Follow along with our experts and explore ideas in quantum programming, quantum machine learning, and quantum chemistry.
```

josh146 commented on index.rst:
```diff
@@ -3,11 +3,27 @@
         }
     </style>
 
-Quantum machine learning
-========================
```
Hey @Anakhom, I'm curious how come you removed the title here in favour of a `h1` down below, something to do with the `<br>`?

I would suggest we move the page heading back up here, mostly because Sphinx uses this to autofill in a lot of other details on the rendered HTML (e.g., the `<title>` tag is now blank 🙁)

Anakhom replied:
@josh146 I see! yeah, I believe at the time I used `<h1>` it to have a better manipulation over the styling of the title, but I will bring this it back.

josh146 commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
-            <br><br>
-            This, in turn, is driving the development of quantum machine learning and variational quantum circuits.</p>
+    <div>
+        <h1 class="mt-6">Welcome to the Quantum learning </br> superhub</h1>
```
@BM7878 What do you think we should do for the page title?

I know we've been using 'superhub' when talking about it, but I'm wondering if it feels a bit too informal.

Options I can think of include:

- Quantum learning
- Welcome to quantum learning
- ?

I think I prefer the first option though!

josh146 replied:
Actually @BM7878 I like your suggestion below, "Learn how to program for quantum computers using PennyLane".

josh146 commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
-                <div class="col-lg-3 mb-5">
+            <div class="row main-cards d-flex justify-content-center">
+                <div class="info-card col-lg-3 mb-5">
                     <a href="whatisqml.html">
```
This we can't update until the QC hub is ready I suppose?

Anakhom replied:
Yep, the change is in the [QC PR](https://github.com/PennyLaneAI/qml/pull/684/files#diff-54a294a5d016e1a8e98bc95668ed84a99a9edd5c10394d9a2b1ee848006e98a7) 🙂

josh146 commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
-                                <img src="_static/flask.png" class="img-fluid" style="max-width: 53px;"></img>
+
+                <div class="info-card col-lg-3 mb-5">
+                    <a href="glossary.html">
```
Just a reminder for us to update this link before merging :)

Anakhom replied:
this is changed in [QML PR](https://github.com/PennyLaneAI/qml/pull/682/files#diff-54a294a5d016e1a8e98bc95668ed84a99a9edd5c10394d9a2b1ee848006e98a7) :slightly_smiling_face:

josh146 commented on index.rst:
```diff
@@ -17,83 +33,74 @@ Quantum machine learning
-                                <img src="_static/video.png" class="img-fluid" style="max-width: 63px;"></img>
+                
+                <div class="info-card col-lg-3 mb-5">
+                    <a href="glossary.html">
```
Just a reminder for us to update this link before merging :)

Anakhom replied:
this is changed in [QChem PR](https://github.com/PennyLaneAI/qml/pull/683/files) 🙂

josh146 commented on index.rst:
```diff
@@ -102,61 +109,108 @@ Quantum machine learning
-    <hr class="pb-5">
+        <section>
+            <div class="d-flex flex-wrap justify-content-center">
+                <h2 class="fw-normal text-center mt-1 mb-5 border-0">Curabitur tempus urna at turpis lobortis. Ut commodo efficitur neque.</h2>
```
```suggestion
                <h2 class="fw-normal text-center mt-1 mb-5 border-0">Browse our full set of demonstrations and tutorials</h2>
```

josh146 commented on index.rst:
```diff
@@ -102,61 +109,108 @@ Quantum machine learning
-                <br>
-                <h5 class="font-weight-bold my-4">Research focused</h5>
+                <div class="info-card col-lg-3 mb-5">
+                    <a href="glossary.html">
```
```suggestion
                    <a href="videos.html">
```

josh146 commented on index.rst:
```diff
@@ -102,61 +109,108 @@ Quantum machine learning
+            </div>
+            <div class="row main-cards d-flex justify-content-center">
+                <div class="info-card col-lg-3 mb-5">
+                    <a href="glossary.html">
```
```suggestion
                    <a href="demonstrations.html">
```

josh146 reviewed (commented):
Thanks @Anakhom! Looking good, this will be ready to merge in once the copy is in and the new hubs/links are in.

Anakhom commented:
@josh146 @BM7878 , thank you for getting back to me with the copy! I have updated the page and you can check the new demo in the description above. 

Please note that links to the three subhubs are not available to add yet and once they are created, I will update them swiftly.

CC @AndrewGardhouse @ashishks0522

Anakhom commented on index.rst:
```diff
@@ -3,11 +3,29 @@
+==========================================================
 
 .. meta::
    :property="og:description": Sit back and learn about the field of quantum machine learning, explore key concepts, download and run QML demonstrations, and view our selection of curated videos.
```
@josh146 @BM7878 Please let me know what would you want in the meta description for the superhub page, along with the image. Thanks!

josh146 replied:
```suggestion
   :property="og:description": Sit back and learn about quantum computing, quantum machine learning, and quantum chemistry by exploring key concepts and working through practical demonstrations.
```

@BM7878 how does that sound? Sounds a bit strange to my ears 🤔

BM7878 replied:
The newer one? It's probably the 'by' that makes it sound a bit strange. I would suggest cutting the sentence into two - perhaps: 

```
Sit back and learn about quantum computing, quantum machine learning, and quantum chemistry. Explore key concepts and work through practical demonstrations.
```

josh146 replied:
```suggestion
   :property="og:description": Sit back and learn about quantum computing, quantum machine learning, and quantum chemistry. Explore key concepts and work through practical demonstrations.
```

josh146 replied:
Thanks @BM7878, @Anakhom this suggestion is good to go

ikurecic commented on index.rst:
```diff
@@ -3,97 +3,106 @@
 
-Quantum machine learning
-========================
+Learn how to program for quantum computers using PennyLane
```
Either 'Learn quantum programming' or 'Learn with PennyLane' (my preference) would work. This is too long.

ikurecic commented on index.rst:
```diff
@@ -3,97 +3,106 @@
+                                Quantum </br> Computing
+                            </h3>
+                            <p class="card-text">
+                            A field of research that explores how universal computation can be achieved through quantum-mechanical systems.
```
```suggestion
                            Learn how universal computation can be achieved using quantum-mechanical systems.
```
Minimal intervention, to fit the other two.

ikurecic commented on index.rst:
```diff
@@ -3,97 +3,106 @@
+                                Quantum </br> Chemistry
+                            </h3>
+                            <p class="card-text">
+                            Study the properties of molecules and materials using quantum computing.
```
```suggestion
                            Study the properties of molecules and materials using quantum computers and algorithms.
```
Adding length and specificity (+algorithms), changing 'computing' to 'computers' to better balance out the other two.

ikurecic commented on index.rst:
```diff
@@ -102,61 +111,106 @@ Quantum machine learning
+                                Videos
+                            </h3>
+                            <p class="card-text">
+                            Follow along with our experts and explore ideas in quantum programming, quantum machine learning, and quantum chemistry.
```
```suggestion
                            Follow along with our experts and explore ideas in quantum computing, quantum machine learning, and quantum chemistry.
```
For consistency?

ikurecic replied:
Also, how do we feel about removing 'quantum' to balance it with the demo link text better?
```suggestion
                            Follow along with our experts and explore ideas in quantum computing, machine learning, and chemistry.
```

ikurecic commented on index.rst:
```diff
@@ -102,61 +111,106 @@ Quantum machine learning
+                <div class="text-center">
+                    <img src="_static/research.png" class="img-fluid"></img>
+                    <br>
+                    <h5 class="font-weight-bold my-4">Research focused</h5>
```
```suggestion
                    <h5 class="font-weight-bold my-4">Research-focused</h5>
```

ikurecic commented on index.rst:
```diff
@@ -102,61 +111,106 @@ Quantum machine learning
+                <div class="text-center">
+                    <img src="_static/fork.png" class="img-fluid"></img>
+                    <br>
+                    <h5 class="font-weight-bold my-4">Open source</h5>
```
```suggestion
                    <h5 class="font-weight-bold my-4">Open-source</h5>
```
Consistency with text

ikurecic commented on index.rst:
```diff
@@ -102,61 +111,106 @@ Quantum machine learning
+                <div class="text-center">
+                    <img src="_static/community.png" class="img-fluid"></img>
+                    <br>
+                    <h5 class="font-weight-bold my-4">Community driven</h5>
```
```suggestion
                    <h5 class="font-weight-bold my-4">Community-driven</h5>
```

ikurecic reviewed (commented):
Just me on my hyphen crusade.

ikurecic commented on index.rst:
```diff
@@ -3,160 +3,214 @@
+                <div class="text-center">
+                    <img src="_static/hubs/research.svg" class="img-fluid" style="max-width: 250px;" alt="Light bulb with an atom inside"></img>
+                    <br>
+                    <h5 class="font-weight-bold my-4">Research-focused</h5>
+                </div>
+                <p class="grey-text mb-0">Xanadu is not just a software company; we also perform high-impact research and build quantum hardware. Our software gets used internally across the company &mdash; check out some of the <a href="https://xanadu.ai/research">papers released</a> using our open-source software tools.
+                </p>
```
This is nitpicking, but the column looks to be really narrow here... and the number of lines is ... a bit much. 😬
Could we either increase the width by the tiniest bit, or crop a few words out of it? For example:

```suggestion
                    <h5 class="font-weight-bold my-4">Research-focused</h5>
                </div>
                <p class="grey-text mb-0">Xanadu is not just a software company; we also perform high-impact research and build quantum hardware. Check out some of the <a href="https://xanadu.ai/research">papers released</a> using our open-source software tools.
                </p>
```

ikurecic reviewed (commented):
Added one comment, but also the vertical space above the footer feels a bit excessive. 😅

Anakhom commented:
> Added one comment, but also the vertical space above the footer feels a bit excessive. 😅

Addressed your comments! No white space between the footer and the section and updated the content.

ikurecic commented on index.rst:
```diff
@@ -171,7 +176,7 @@ Learn quantum programming
                     <h5 class="font-weight-bold my-4">Research-focused</h5>
                 </div>
-                <p class="grey-text mb-0">Xanadu is not just a software company; we also perform high-impact research and build quantum hardware. Our software gets used internally across the company &mdash; check out some of the <a href="https://xanadu.ai/research">papers released</a> using our open-source software tools.
+                <p class="grey-text mb-0">Xanadu is not just a software company; we also perform high-impact research and build quantum hardware. Check out some of the <a href="https://xanadu.ai/research">papers released</a> using our open-source software tools.
```
@josh146 how do you feel about this change?

josh146 commented on index.rst:
```diff
@@ -1,162 +1,221 @@
     </div>
-    <!-- Grid row -->
 
 .. toctree::
     :maxdepth: 2
```
@Anakhom only minor, but the toctree will need to be updated once the subhubs are merged but before this is merged :)

ashishks0522 reviewed (approved):
Everything together looks great! Thanks @Anakhom 💯

josh146 commented on index.rst:
```diff
@@ -1,168 +1,200 @@
+                    <br>
+                    <h5 class="font-weight-bold my-4">Community-driven</h5>
+                </div>
+                    <p class="grey-text mb-0">When we release software, we have the community in mind. Development takes place publicly on GitHub, and members of our team are available to chat on our <a href="https://u.strawberryfields.ai/slack">Strawberry Fields Slack</a> and <a href="https://discuss.pennylane.ai">PennyLane forum</a>.
```
```suggestion
                    <p class="grey-text mb-0">When we release software, we have the community in mind. Development takes place publicly on GitHub, and members of our team are available to chat on our <a href="https://u.strawberryfields.ai/slack">Slack</a> and <a href="https://discuss.pennylane.ai">discussion forum</a>.
```

josh146 reviewed (approved):
Amazing work! One small copy change but everything else looks great
